package/kedro_viz/data_access/repositories/modular_pipelines.py

```diff
@@ -237,7 +247,26 @@ def has_modular_pipeline(self, modular_pipeline_id: str) -> bool:
             False
         """
         return modular_pipeline_id in self.tree
+    
+    def get_modular_pipeline_for_node(self, node) -> Optional[str]:
+        """Get the name of the modular pipeline to which the given node belongs.
+
+        Args:
+            node: The node to check for its parent modular pipeline.
+
+        Returns:
+            The name of the modular pipeline if the node belongs to any modular pipeline,
+            otherwise returns None.
+        """
+
+        node_id = GraphNode._hash(str(node))

```

Comment:
I am not sure if this is WIP, but this will be expensive for each node. It would be nice to have a map like -
`{nodeId : modularPipelineId}` and if we need to know which registerd pipeline as well, we can have a map like -

```python
{nodeId: {"modular_pipeline": mod_pipeline_id, "registerd_pipeline": reg_pipeline_id}} 

# OR

{node_id: (mod_pipeline_id, reg_pipeline_id)}
```

Reply:
I also saw we are now having modular_pipeline under GraphNode schema. Is this some test function ?